bot/src/main/java/com/intern/bot/config/TelegramConfig.java

```diff
@@ -12,17 +13,22 @@
 @Slf4j
 @RequiredArgsConstructor
 @Configuration
+@EnableConfigurationProperties(BotProperties.class)
 public class TelegramConfig {
 
-  private final BotProperties bot;
+  private final TelegramBot bot;
 
   @Bean
-  public void connect() {
+  public TelegramBotsApi registerBot() {
+    TelegramBotsApi botsApi = null;
     try {
-      TelegramBotsApi botsApi = new TelegramBotsApi(DefaultBotSession.class);
+      botsApi = new TelegramBotsApi(DefaultBotSession.class);
       botsApi.registerBot(bot);
     } catch (TelegramApiException e) {
-      log.error(e.getMessage());
+      log.error("Could not initialize telegram bot API.", e);
```

Comment:
Your bot won't work if your main API fails to initialise. So you need to let this bean throw the exception upward, instead of just logging it.